src_c/display.c

```
@@ -1334,7 +1259,8 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
 
     /*set the window icon*/
     if (!state->icon) {
-        state->icon = pg_display_resource(icon_defaultname);
+        PyObject *icon_path_obj = pg_GetPkgdataResource(icon_defaultname);
+        state->icon = pgImage_LoadBasic(icon_path_obj);
```

same as above review (except here maybe just ignore the error and don't even call `pgImage_LoadBasic` in that case)